src/main/java/io/quarkus/search/app/WebComponents.java

```diff
@@ -0,0 +1,35 @@
+package io.quarkus.search.app;
+
+import io.quarkiverse.web.bundler.runtime.Bundle;
+import jakarta.inject.Inject;
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.core.Context;
+import jakarta.ws.rs.core.Response;
+import jakarta.ws.rs.core.UriBuilder;
+import jakarta.ws.rs.core.UriInfo;
+import org.jboss.resteasy.reactive.Cache;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+
+@Path("/web-components")
+public class WebComponents {
+
+    @Inject
+    Bundle bundle;
+
+    @Context
+    UriInfo uriInfo;
+
+    // This route allows to access the web-component js on a fixed path
+    // without affecting caching of the script
+    @Path("/search.js")
+    @Cache(noCache = true)
```

Comment:
As discussed on Zulip, only the redirect will not be cached (the script will be cached), so that if the script changes, the the redirect will change (and be called again).

Also as discussed there is a price for this but I am not sure there is a way to know if it has changed without any call?

Reply:
> As discussed on Zulip, only the redirect will not be cached (the script will be cached), so that if the script changes, the the redirect will change (and be called again).

Seems fair.

> Also as discussed there is a price (on REST call) for this but I am not sure there is a way to know if it has changed without any call?

Makes sense.